static/js/src/advantage/subscribe/listeners/ui-event-listeners.js

```diff
@@ -1,16 +1,28 @@
-import { toggleOtherVersionsModal } from "../reducers/ui-reducer";
+import {
+  toggleOtherVersionsModal,
+  togglePurchaseModal,
+} from "../reducers/ui-reducer";
 
 const form = document.querySelector(".js-shop-form");
 
 export default function initUIControls(store) {
   const otherVersionToggles = form.querySelectorAll(
     "[aria-controls='other-versions-modal']"
   );
-
   otherVersionToggles.forEach((toggle) => {
     toggle.addEventListener("click", (e) => {
       e.preventDefault();
       store.dispatch(toggleOtherVersionsModal());
     });
   });
+
+  const purchaseModalToggles = document.querySelectorAll(
+    "[aria-controls='purchase-modal']"
+  );
+  purchaseModalToggles.forEach((toggle) => {
+    toggle.addEventListener("click", (e) => {
```

Comment:
As this is an exported function, how do we know that we won't get multiple events attached?

Reply:
Currently, `initUIControls` is only called once.
I could switch from `.addEventListener(...)` to `.onClick=(...)` to make sure we don't attach duplicate event listeners. Would that be fine or do you know of a better way?